pkg/nsx/services/vpc/vpc.go

```diff
@@ -1005,50 +757,82 @@ func (s *VPCService) GetNSXLBSPath(lbsId string) string {
 	return *vpcLBS.Path
 }
 
+func (vpcService *VPCService) EdgeClusterEnabled(nc *common.VPCNetworkConfigInfo) bool {
+	var vpcConnectivityProfile *model.VpcConnectivityProfile
+	var getErr error
+	if err := retry.OnError(retry.DefaultBackoff, func(err error) bool {
+		if err == nil {
+			return false
+		}
+		_, errortype := nsxutil.DumpAPIError(err)
+		if *errortype == apierrors.ErrorType_SERVICE_UNAVAILABLE || *errortype == apierrors.ErrorType_TIMED_OUT {
+			return true
+		} else {
+			return false
+		}
+	}, func() error {
+		vpcConnectivityProfile, getErr = vpcService.GetVpcConnectivityProfile(nc, nc.VPCConnectivityProfile)
+		return getErr
+	}); err == nil {
+		log.Info("vpc connectivity profile", "service gateway enable", *vpcConnectivityProfile.ServiceGateway.Enable)
+		return *vpcConnectivityProfile.ServiceGateway.Enable
+	} else {
+		log.Error(getErr, "failed to get vpc connectivity profile", "vpc connectivity profile", nc.VPCConnectivityProfile)
+		return false
+	}
+
+}
 func GetAlbEndpoint(cluster *nsx.Cluster) error {
 	_, err := cluster.HttpGet(albEndpointPath)
 	return err
 }
 
-func (vpcService *VPCService) NSXLBEnabled() bool {
+func (vpcService *VPCService) GetLBProvider() LBProvider {
 	lbProviderMutex.Lock()
 	defer lbProviderMutex.Unlock()
-
-	if lbProvider == "" {
-		lbProvider = vpcService.getLBProvider()
+	if globalLbProvider != NoneLB {
+		log.V(1).Info("lb provider", "current provider", globalLbProvider)
+		return globalLbProvider
 	}
-	return lbProvider == LBProviderNSX
-}
 
-func (vpcService *VPCService) getLBProvider() string {
+	ncName := common.SystemVPCNetworkConfigurationName
+	netConfig, _ := vpcService.GetVPCNetworkConfig(ncName)
+	nc := &netConfig
+
+	edgeEnable := vpcService.EdgeClusterEnabled(nc)
```

Comment:
Just need to check CR status about AutoSnatEnabled

Reply:
add checking